File: Raven.Abstractions/Commands/PutCommandData.cs
//-----------------------------------------------------------------------
// <copyright file="PutCommandData.cs" company="Hibernating Rhinos LTD">
//     Copyright (c) Hibernating Rhinos LTD. All rights reserved.
// </copyright>
//-----------------------------------------------------------------------
using Raven.Abstractions.Data;
using Raven.Json.Linq;

namespace Raven.Abstractions.Commands
{
    /// <summary>
    /// A single batch operation for a document PUT
    /// </summary>
    public class PutCommandData : ICommandData
    {
        /// <summary>
        /// Key of a document.
        /// </summary>
        public virtual string Key { get; set; }

        /// <summary>
        /// Returns operation method. In this case PUT.
        /// </summary>
        public string Method
        {
            get { return "PUT"; }
        }

        /// <summary>
        /// Current document etag, used for concurrency checks (null to skip check)
        /// </summary>
        public virtual Etag Etag { get; set; }

        /// <summary>
        /// RavenJObject representing the document.
        /// </summary>
        public virtual RavenJObject Document { get; set; }

        /// <summary>
        /// Information used to identify a transaction. Contains transaction Id and timeout.
        /// </summary>
        public TransactionInformation TransactionInformation { get; set; }

        /// <summary>
        /// RavenJObject representing document's metadata.
        /// </summary>
        public virtual RavenJObject Metadata { get; set; }

        /// <summary>
        /// Additional command data. For internal use only.
        /// </summary>
        public RavenJObject AdditionalData { get; set; }

        /// <summary>
        /// Translates this instance to a Json object.
        /// </summary>
        /// <returns>RavenJObject representing the command.</returns>
        public RavenJObject ToJson()
        {
            var ret = new RavenJObject
            {
                {"Key", Key},
                {"Method", Method},
                {"Document", Document},
                {"Metadata", Metadata},
                {"AdditionalData", AdditionalData}
            };
            if (Etag != null)
                ret.Add("Etag", Etag.ToString());
            return ret;
        }
    }
}

File: Raven.Client.Java/src/main/java/net/ravendb/abstractions/commands/PutCommandData.java
package net.ravendb.abstractions.commands;

import net.ravendb.abstractions.data.Etag;
import net.ravendb.abstractions.data.HttpMethods;
import net.ravendb.abstractions.json.linq.RavenJObject;
import net.ravendb.abstractions.json.linq.RavenJValue;

/**
 *  A single batch operation for a document PUT
 */
public class PutCommandData implements ICommandData {
  private String key;
  private Etag etag;
  private RavenJObject document;
  private RavenJObject metadata;
  private RavenJObject additionalData;


  public PutCommandData() {
    super();
  }

  public PutCommandData(String key, Etag etag, RavenJObject document, RavenJObject metadata) {
    super();
    this.key = key;
    this.etag = etag;
    this.document = document;
    this.metadata = metadata;
  }

  @Override
  public RavenJObject getAdditionalData() {
    return additionalData;
  }

  public RavenJObject getDocument() {
    return document;
  }

  @Override
  public Etag getEtag() {
    return etag;
  }

  @Override
  public String getKey() {
    return key;
  }

  @Override
  public RavenJObject getMetadata() {
    return metadata;
  }

  @Override
  public HttpMethods getMethod() {
    return HttpMethods.PUT;
  }

  @Override
  public void setAdditionalData(RavenJObject additionalData) {
    this.additionalData = additionalData;
  }

  public void setDocument(RavenJObject document) {
    this.document = document;
  }

  public void setEtag(Etag etag) {
    this.etag = etag;
  }

  public void setKey(String key) {
    this.key = key;
  }

  public void setMetadata(RavenJObject metadata) {
    this.metadata = metadata;
  }

  @Override
  public RavenJObject toJson() {
    RavenJObject value = new RavenJObject();
    value.add("Key", new RavenJValue(key));
    value.add("Method", new RavenJValue(getMethod().name()));
    value.add("Document", document);
    value.add("Metadata", metadata);
    value.add("AdditionalData", additionalData);

    if (etag != null) {
      value.add("Etag", new RavenJValue(etag.toString()));
    }

    return value;
  }

}
